### PR TITLE
Manager cleanup

### PR DESCRIPTION
**Pull request by pd0wm**

### CODE REVIEW

File: selfdrive/manager/manager.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    params.put("Passive", str(int(os.getenv("PASSIVE"))))
+
+  if params.get("Passive") is None:
+    raise Exception("Passive must be set to continue")
+
+  if EON:
+    update_apks()
+
+  manager_init(spinner)
+  manager_prepare(spinner)
```
Comment (adeebshihadeh) [inline selfdrive/manager/manager.py:178]: merge these? unclear how these are different

Comment (adeebshihadeh) [reply selfdrive/manager/manager.py:178]: the top half of this main function is like a third init too

File: selfdrive/manager/process.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  # exec the process
+  os.chdir(cwd)
+  os.execvp(pargs[0], pargs)
+
+
+def join_process(process, timeout):
+  # Process().join(timeout) will hang due to a python 3 bug: https://bugs.python.org/issue28382
+  # We have to poll the exitcode instead
+  t = time.time()
+  while time.time() - t < timeout and process.exitcode is None:
```
Comment (adeebshihadeh) [inline selfdrive/manager/process.py:52]:
```suggestion
  t = time.monotonic()
  while time.monotonic() - t < timeout and process.exitcode is None:
```

File: selfdrive/test/test_manager.py [hunk 3]
```diff
@@ -23,37 +26,44 @@ def tearDown(self):
 
   def test_manager_prepare(self):
     os.environ['PREPAREONLY'] = '1'
-    manager.main()
+
+    with Spinner() as spinner:
```
Comment (adeebshihadeh) [inline selfdrive/test/test_manager.py:30]: can't it just default to none? or maybe make it's own spinner?

**Comment by adeebshihadeh** [comment]
Let's also move the manager test to selfdrive/manager/

File: selfdrive/manager/process.py (hunk 2 above)
Comment (pd0wm) [reply selfdrive/manager/process.py:52]: Maybe with python 3.8 this whole function can go. Will look into that after merging.

File: selfdrive/manager/manager.py (hunk 1 above)
Comment (pd0wm) [reply selfdrive/manager/manager.py:178]: Init and prepare are different. prepare calls prepare on all the proceses, but doesn't do anything to the state in manager.

**Pull request merged**